internals/secrethub/credential_store.go

```diff
@@ -40,17 +40,21 @@ func NewCredentialStore(io ui.IO) CredentialStore {
 }
 
 type credentialStore struct {
-	ConfigDir                    string
+	configDir                    ConfigDir
 	AccountCredential            string
 	credentialPassphrase         string
 	CredentialPassphraseCacheTTL time.Duration
 	credential                   *secrethub.RSACredential
 	io                           ui.IO
 }
 
+func (store *credentialStore) ConfigDir() configdir.Dir {
+	return store.configDir.Dir
+}
+
 // Register registers the flags for configuring the store on the provided Registerer.
 func (store *credentialStore) Register(r FlagRegisterer) {
-	r.Flag("config-dir", "The absolute path to a custom configuration directory. Defaults to $HOME/.secrethub").StringVar(&store.ConfigDir)
+	r.Flag("config-dir", "The absolute path to a custom configuration directory. Defaults to $HOME/.secrethub").Default("").SetValue(&store.configDir)
```

Comment:
Could we omit the default and initialize the config directory to the default value in `NewCredentialStore` instead (ideally using a `NewConfigDirFlag` function)?

Reply:
Let's not do this. The default will then always be evaluated. When the home directory cannot be determined, no error should be returned when the home directory cannot be determined, as we don't need it then.

Reply:
Decided to leave it in for now.
64620dea4a9be66ee4253e5f99a7cafbd04b177d fixes the default value being displayed in the help text.